### DISCUSSION TITLE
Trying to use with 4.2 and failing.

### DISCUSSION THREAD

**Discussion by HungryCatGames** [TokisanGames/Terrain3D#262] [Questions] [answered]
I followed (or at least tried to) the instructions for installation and got a bunch of errors on first load. I tried running anyway and capsule appears and then goes into free fall. I quit. Launched again, same errors and tried to enable the plugin and I get:

Unable to load addon script from path: 'res://addons/terrain_3d/editor/editor.gd'. This might be due to a code error in that script.
Disabling the addon at 'res://addons/terrain_3d/plugin.cfg' to prevent further errors.

The console errors being reported are:

Godot Engine v4.2.stable.official.46dc27791 - https://godotengine.org
OpenGL API 3.3.0 NVIDIA 471.41 - Compatibility - Using Device: NVIDIA - NVIDIA GeForce RTX 3080

Editing project: D:/Godot App/Terrain3D-main/project
ERROR: GDExtension library not found:
   at: open_library (core/extension/gdextension.cpp:688)
ERROR: Failed loading resource: res://addons/terrain_3d/terrain.gdextension. Make sure resources have been imported by opening the project in the editor at least once.
   at: (core/io/resource_loader.cpp:274)
ERROR: Error loading extension: res://addons/terrain_3d/terrain.gdextension
   at: (core/extension/gdextension_manager.cpp:234)
Godot Engine v4.2.stable.official.46dc27791 - https://godotengine.org
Vulkan API 1.2.175 - Forward+ - Using Vulkan Device #0: NVIDIA - NVIDIA GeForce RTX 3080

WARNING: Blend file import is enabled in the project settings, but no Blender path is configured in the editor settings. Blend files will not be imported.
     at: _editor_init (modules/gltf/register_types.cpp:63)
SCRIPT ERROR: Parse Error: Could not find type "Terrain3D" in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:11)
SCRIPT ERROR: Parse Error: Could not find type "Terrain3DEditor" in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:13)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DEditor" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:24)
SCRIPT ERROR: Parse Error: Could not find type "Terrain3D" in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:49)
SCRIPT ERROR: Parse Error: Could not find type "Terrain3D" in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:56)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DEditor" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:139)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DEditor" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:142)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DStorage" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:143)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DEditor" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:144)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DStorage" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:145)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DEditor" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:146)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DEditor" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:152)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DEditor" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:156)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DEditor" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:160)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DEditor" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:161)
SCRIPT ERROR: Parse Error: Could not find type "Terrain3DTexture" in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:187)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DTextureList" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:190)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DEditor" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:199)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DEditor" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:200)
ERROR: Failed to load script "res://addons/terrain_3d/editor/editor.gd" with error "Parse error".
   at: load (modules/gdscript/gdscript.cpp:2775)
ERROR: GDExtension library not found:
   at: open_library (core/extension/gdextension.cpp:688)
ERROR: Failed loading resource: res://addons/terrain_3d/terrain.gdextension. Make sure resources have been imported by opening the project in the editor at least once.
   at: (core/io/resource_loader.cpp:274)
ERROR: Cannot get class 'Terrain3D'.
   at: (core/object/class_db.cpp:358)
SCRIPT ERROR: Parse Error: Could not find type "Terrain3D" in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:11)
SCRIPT ERROR: Parse Error: Could not find type "Terrain3DEditor" in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:13)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DEditor" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:24)
SCRIPT ERROR: Parse Error: Could not find type "Terrain3D" in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:49)
SCRIPT ERROR: Parse Error: Could not find type "Terrain3D" in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:56)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DEditor" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:139)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DEditor" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:142)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DStorage" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:143)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DEditor" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:144)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DStorage" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:145)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DEditor" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:146)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DEditor" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:152)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DEditor" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:156)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DEditor" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:160)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DEditor" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:161)
SCRIPT ERROR: Parse Error: Could not find type "Terrain3DTexture" in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:187)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DTextureList" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:190)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DEditor" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:199)
SCRIPT ERROR: Parse Error: Identifier "Terrain3DEditor" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:200)
ERROR: Failed to load script "res://addons/terrain_3d/editor/editor.gd" with error "Parse error".
   at: load (modules/gdscript/gdscript.cpp:2775)

Is this a compatibility issue with 4.2? Or am I doing something wrong? I downloaded the ZIP today so I am assuming it is current and didn't see any other discussions on this issue,

**Comment by TokisanGames** [comment]
4.2 is supported. 

```Editing project: D:/Godot App/Terrain3D-main/project
ERROR: GDExtension library not found:
```

Did you download the source code and then not compile it, instead of downloading a binary release? Read through the directions again and either download the build for version 4.2 linked in the release notes, or compile it yourself.

There's also a whole troubleshooting section in the docs. 100% chance you didn't install it properly. Your error says it can't find the library on your system. So if it can't find it, it can't load it.

**Reply by TokisanGames** [reply, marked as answer]
Closed due to no response and not reproducible. It can be reopened if you wish to troubleshoot. Watch the youtube tutorial for help installing properly. There's also a new version released today to download.

https://youtu.be/oV8c9alXVwU
